src/app/state/playback_state.rs

```diff
@@ -310,10 +316,19 @@ impl PlaybackState {
 
     fn play_next(&mut self) -> Option<String> {
         let len = self.running_order().len();
-        let next = self
-            .position
-            .filter(|&p| p.index + 1 < len)
-            .map(|p| p.index + 1);
+        let next = self.position.and_then(|p|{
+            match self.repeat {
```

Comment:
bit better already !! :) 

i had something like that in mind (havent tested)

```rust
match self.repeat {
    RepeatMode::Song => Some(p.index),
    RepeatMode::Playlist  => Some((p.index + 1)%len),
    RepeatMode::None => Some(p.index + 1).filter(|i| i < len),
}
```

Reply:
Yes your code is correct with some tweaking :)